### DISCUSSION TITLE
how to build pipelines?

### DISCUSSION THREAD

**Discussion by arron-lyu** [duckdb/duckdb#3160] [Q&A]
Hello!
DuckDB is greate! I read the code these days, but I don't understand the function of `Executor::BuildPipelines(PhysicalOperator *op, Pipeline *current)`. How to build pipelines? what is the theoretical basis？which operators should be put into one pipeline? 
Thanks!

**Comment by Alex-Monahan** [comment]
Hello! 
I don't have the expertise to directly answer your question, but have you seen this presentation by Mark? It describes DuckDB's new [push based execution model](https://youtu.be/MA0OsvYFGrc) and might help with your question!
